detectable/src/main/java/com/synopsys/integration/detectable/detectables/pip/PipInspectorDetectable.java

```diff
@@ -69,12 +76,21 @@ public PipInspectorDetectable(final DetectableEnvironment environment, final Fil
     @Override
     public DetectableResult applicable() {
         setupFile = fileFinder.findFile(environment.getDirectory(), SETUPTOOLS_DEFAULT_FILE_NAME);
-        final boolean hasSetups = setupFile != null;
-        final boolean hasRequirements = pipInspectorDetectableOptions.getRequirementsFilePaths() != null && pipInspectorDetectableOptions.getRequirementsFilePaths().size() > 0;
+        boolean hasSetups = setupFile != null;
+
+        requirementsFiles = pipInspectorDetectableOptions.getRequirementsFilePaths();
+        if (CollectionUtils.isEmpty(pipInspectorDetectableOptions.getRequirementsFilePaths())) {
+            requirementsFiles = fileFinder.findFiles(environment.getDirectory(), REQUIREMENTS_DEFAULT_FILE_NAME)
+                                    .stream()
+                                    .map(File::toPath)
+                                    .collect(Collectors.toList());
+        }
+        boolean hasRequirements = CollectionUtils.isNotEmpty(requirementsFiles);
```

Comment:
Any reason for not using `requirementFiles.isEmpty()`?

Reply:
Previous code checked if the requirementsFiles object was null, so I assumed it must be nullable.

Reply:
I suppose at this point we're no longer nullable, so we could use the other method but meh-- I think isNotEmpty tells a better story than !object.isEmpty()

Reply:
I agree with that